python/bindings.cpp

```diff
@@ -589,6 +590,109 @@ PYBIND11_MODULE(_libsonata, m) {
         .value("center", SimulationConfig::Report::Compartments::center)
         .value("all", SimulationConfig::Report::Compartments::all);
 
+    py::class_<SimulationConfig::Input> input(m, "Input", "List of parameters of an input");
+    input.def_readonly("module", &SimulationConfig::Input::module, "Type of stimulus")
+        .def_readonly("input_type", &SimulationConfig::Input::input_type, "Type of input")
+        .def_readonly("delay", &SimulationConfig::Input::delay, "Time when input is activated (ms)")
+        .def_readonly("duration",
+                      &SimulationConfig::Input::duration,
+                      "Time duration in ms for how long input is activated")
+        .def_readonly("node_set",
+                      &SimulationConfig::Input::node_set,
+                      "Node set which is affected by input")
+        .def_readonly("amp_start",
+                      &SimulationConfig::Input::amp_start,
+                      "The amount of current initially injected (nA)")
+        .def_readonly("amp_end",
+                      &SimulationConfig::Input::amp_end,
+                      "The final current when a stimulus concludes (nA)")
+        .def_readonly("percent_start",
+                      &SimulationConfig::Input::percent_start,
+                      "The percentage of a cell's threshold current to inject ")
+        .def_readonly("percent_end",
+                      &SimulationConfig::Input::percent_end,
+                      "The percentage of a cell's threshold current to inject at the end")
+        .def_readonly("width",
+                      &SimulationConfig::Input::width,
+                      "The length of time each pulse lasts (ms)")
+        .def_readonly("frequency",
+                      &SimulationConfig::Input::frequency,
+                      "The frequency of pulse trains (Hz)")
+        .def_readonly("percent_less",
+                      &SimulationConfig::Input::percent_less,
+                      "A percentage adjusted from 100 of a cell's threshold current")
+        .def_readonly("spike_file",
+                      &SimulationConfig::Input::spike_file,
+                      "The location of the file with the spike info for injection")
+        .def_readonly("source",
+                      &SimulationConfig::Input::source,
+                      "The node set to replay spikes from")
+        .def_readonly("voltage",
+                      &SimulationConfig::Input::voltage,
+                      "The membrane voltage the targeted cells should be held at (mV)")
+        .def_property_readonly(
+            "mean",
+            [](const SimulationConfig::Input& input) -> nonstd::optional<double> {
+                if (input.mean > std::numeric_limits<double>::lowest())
```

Comment:
We intend to use this lowest value as (invalid) default in case that the key is absence in the config, and in python we can return a `NoneType`. I don't know how to name it property ? `InvalidDefault` ? or any proposal ?
In this case, I think that I could also use `-1 `as default because the current is always `>= 0`.  I have no preference.

Reply:
Ah, I see, I missed that.
What about if `Input::mean` and `Input::meanPercent` were `nonstd::optional` and only the correct one gets set?

Reply:
Ah, yes, sounds a good idea! Let me try.

Reply:
I just pushed https://github.com/BlueBrain/libsonata/pull/196/commits/91f131208b4172a5f6adb39816024f839a22918c
Is that what you meant ? We just need to add a json serializer to convert from/to `nonstd::optional`

Reply:
Yup!  You even got fancy w/ the `adl_serializer`; looks good to me.